### ISSUE TITLE
option --else for test is ignored

### ISSUE THREAD

**Issue by jbpotonnier** [purescript/spago#630]
When running tests with `spago test -w  --else 'notify-send Failed'`, nothing happens when a test is failing.

The `--then` option is working properly though.

The version is `0.15.2`

**Comment by f-f** [comment]
@jbpotonnier thanks for reporting!

You found a very interesting corner case - let me walk you through how the `before`/`else`/`then` hooks are implemented. This is the code in question:

https://github.com/purescript/spago/blob/7391b54ae27189f9414a4962cafce91cbab84eb6/src/Spago/Build.hs#L87-L91

A bit of context: in here the `buildBackend` is basically `purs build`, and `maybePostBuild` is basically `node .spago/run.js` (which contains the tests to be run - that is, `spago test` is just a special case of `spago run`)

At a first look this code is perfectly legit, but if we squint harder a line 89 we can see that the `else` only runs when *the build* fails, and it doesn't when *the tests* fail! (which are run in line 90 all by themselves)

So I think the fix should be something like this, where we run `else` if either the build or the result of the run fail:
```haskell
      buildAction globs = do
        let action = buildBackend globs >> fromMaybe (pure ()) maybePostBuild
        runCommands "Before" beforeCommands
        action `onException` (runCommands "Else" elseCommands)
        runCommands "Then" thenCommands
```

Makes sense?

**Comment by jbpotonnier** [comment]
Hi fabrizio,
I think it does make sense, having your explanation.
I made this mistake in the CI build of one of my former professional project.

I was asking myself how to prevent this, in the model or maybe by automated test.
I don't know the code base though, and I'm just starting Purescript. But coming from Haskell the tools and language is a joy to work with.

**Comment by f-f** [comment]
@jbpotonnier no worries! Fix for this is up in #632
